packages/eslint-plugin/tests/lib/rules/boolean-literal-compare.js

```diff
@@ -0,0 +1,203 @@
+/**
+ * @fileoverview Flags unnecessary equality comparisons against booleans
+ * @author Josh Goldberg <https://github.com/joshuakgoldberg>
+ */
+'use strict';
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require('../../../lib/rules/boolean-literal-compare'),
+  RuleTester = require('eslint').RuleTester,
+  path = require('path');
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const rootDir = path.join(process.cwd(), 'tests/fixtures');
+const parserOptions = {
+  tsconfigRootDir: rootDir,
+  project: './tsconfig.json'
+};
+const ruleTester = new RuleTester({
+  parserOptions,
+  parser: '@typescript-eslint/parser'
+});
+
+ruleTester.run('boolean-literal-compare', rule, {
+  valid: [
+    `
+declare const varAny: any;
+declare const varString: string;
+declare const varObject: {};
+
+varAny === 'true';
```

Comment:
Please separate test cases.

Reply:
@mysticatea although I'm up for separating test cases, it seems a little unnecessary given the upcoming #77 test improvements, as the tests here are closer to how they'd look with that system. I'm a little hesitant to add extra work soon. Would you still like me to separate the test cases?

Reply:
Oh, I didn't realize #77. In that case, we can wait for the PR.

Reply:
we are not sure if #77 will be implemented or if so when. i think its better to keep tests separated for now.

Reply:
Ok, split them up. 😢